cachemonster/services/CacheMonster_ExternalCloudFlareService.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace Craft;
+
+/**
+ * Class CacheMonster_ExternalVarnishService
+ *
+ * @package   CacheMonster
+ * @author    Josh Angell
+ * @copyright Copyright (c) 2016, Supercool Ltd
+ * @link      http://plugins.supercooldesign.co.uk
+ */
+
+class CacheMonster_ExternalCloudFlareService extends BaseApplicationComponent implements ICacheMonster_External
+{
+	/**
+	 * @var string
+	 */
+	private static $_cloudFlareApiBaseUrl = 'https://api.cloudflare.com/client/v4/zones/';
+
+	/**
+	 * Purges the given paths
+	 *
+	 * @param array $paths An array of paths to purge
+	 *
+	 * @return bool
+	 */
+	public function purgePaths($paths=array())
+	{
+		if (!$this->_hasValidSettings())
+		{
+			throw new Exception(Craft::t('Could not validate the CloudFlare cache settings! Please ensure you have entered an "authEmail", "authKey" and "zoneId".'));
+		}
+
+		$settings = craft()->config->get('externalCachingServiceSettings', 'cacheMonster');
+
+		// Normalize the paths returned from Craft's cache table
+		// by removing "site:"" and appending the site url
+		$preparedPaths = $this->_preparePaths($paths);
+
+		// Make the client
+		$client = new \Guzzle\Http\Client();
+
+		// Set the relevant headers
+		$client->setDefaultOption('headers', array(
+			'X-Auth-Email' => $settings['authEmail'],
+			'X-Auth-Key'   => $settings['authKey'],
+			'Content-Type' => 'application/json'
+		));
+
+		try
+		{
+			// Issue a DELETE request to the purge endpoint passing
+			// along the list of files to be purged
+			$url = $this->_purgeUrlForZone($settings['zoneId']);
+			$body = json_encode(array('files' => $preparedPaths));
+			$request = $client->delete($url);
+			$request->setBody($body);
+			$request->send();
+		}
+		catch (\Exception $e)
+		{
+			CacheMonsterPlugin::log('An exception occurred: '.$e->getMessage(), LogLevel::Error);
+			return false;
+		}
+
+		// Just pretend it always worked
```

Comment:
Yep - this is because when using with Tasks, if you return false the Task will error and stop - typically we wouldn’t want it to do that as there could be other steps in the Task that will work just fine.